spec/01-unit/31-simdjson/02-yield_spec.lua

```diff
@@ -0,0 +1,107 @@
+local orig_ngx_sleep = ngx.sleep
+
+
+local spy_ngx_sleep
+local simdjson
+local test_obj
+local test_str
+
+
+describe("[yield] ", function ()
+  lazy_setup(function()
+    test_obj = { str = string.rep("a", 2100), }
+
+    local arr = {}
+    for i = 1, 1000 do
+      arr[i] = i
+    end
+
+    test_str = "[" .. table.concat(arr, ",") .. "]"
+  end)
+
+
+  before_each(function()
+    spy_ngx_sleep = spy.on(ngx, "sleep")
+    simdjson = require("resty.simdjson")
+  end)
+
+
+  after_each(function()
+    ngx.sleep = orig_ngx_sleep  -- luacheck: ignore
+    package.loaded["resty.simdjson"] = nil
+    package.loaded["resty.simdjson.decoder"] = nil
+    package.loaded["resty.simdjson.encoder"] = nil
+  end)
+
+
+  it("enabled when encoding", function()
```

Comment:
We can use a loop to iterate over 4 combinations

Reply:
good suggestion, done.